### PR TITLE
fix: add reset and branch params for indexes

### PR DESCRIPTION
**Pull request by dciborow**

### CODE REVIEW

File: tests/test_gpt_cli.py [hunk 1]
```diff
@@ -85,11 +82,9 @@ class CLICase2(CLICase):
...
 """,
     ),
-    CLICase(f"ask --files {SAMPLE_FILE} --files {SAMPLE_FILE} what programming language is this code written in?"),
-    CLICase(f"ask --fast -f {SAMPLE_FILE} what programming language is this code written in?"),
-    CLICase(f"ask --files src/gpt_review/__init__.py --files src/gpt_review/__init__.py {WHAT_LANGUAGE}"),
-    CLICase(f"ask --fast -f src/gpt_review/__init__.py {WHAT_LANGUAGE}"),
-    CLICase(f"ask --fast -d src/gpt_review --recursive --hidden --required-exts .py {WHAT_LANGUAGE}"),
+    CLICase(f"ask --files {SAMPLE_FILE} --files {SAMPLE_FILE} {WHAT_LANGUAGE} --refresh"),
+    CLICase(f"ask --fast -f {SAMPLE_FILE} {WHAT_LANGUAGE}"),
+    CLICase(f"ask --fast -d src/gpt_review --refresh --recursive --hidden --required-exts .py {WHAT_LANGUAGE}"),
```
Comment (danay1999) [inline tests/test_gpt_cli.py:87]: The tests for --branch seem to be missing.

**Review by deepika087 — APPROVED** [review]

File: tests/test_gpt_cli.py [hunk 2]
```diff
@@ -85,11 +82,9 @@ class CLICase2(CLICase):
...
     CLICase(f"ask --fast -repo microsoft/gpt-review {WHAT_LANGUAGE}"),
```
Comment (dciborow) [inline tests/test_gpt_cli.py:88]:
```suggestion
    CLICase(f"ask --fast -repo microsoft/gpt-review --branch main {WHAT_LANGUAGE}"),
```

File: src/gpt_review/_ask.py [hunk 3]
```diff
@@ -252,8 +258,15 @@ def load_arguments(loader: CLICommandsLoader) -> None:
...
-                help="Refresh the index. Requires --directory, --files, or --repository.",
+                "branch",
+                type=str,
+                help="Branch to index. Default: main.",
+                default="main",
+                options_list=("--branch", "-b"),
+            )
+            args.argument(
+                "reset",
+                help="Rest the index by overwriting the storage directory. Requires --directory, --files, or --repository.",
```
Comment (dciborow) [inline src/gpt_review/_ask.py:269]:
```suggestion
                help="Reset the index, overwriting the directory. Requires --directory, --files, or --repository.",
```

File: src/gpt_review/_ask.py [hunk 4]
```diff
@@ -92,6 +95,7 @@ def _ask(
         fast (bool, optional): Use the fast model. Defaults to False.
         large (bool, optional): Use the large model. Defaults to False.
         directory (Optional[str], optional): The directory to search. Defaults to None.
+        reset (bool, optional): Overwrite the existing storage directory.
```
Comment (dciborow) [inline src/gpt_review/_ask.py:98]:
```suggestion
        reset (bool, optional): Whether to reset the index. Defaults to False.
```

**Pull request merged**